### PR TITLE
[Merged by Bors] - feat: port Topology.ContinuousFunction.Bounded

### PR DESCRIPTION
**Pull request by int-y1**
---

todo:

* [x] instance `CompleteSpace`, find an alternative to `by mono*`
  A: Used a couple of `add_le_add`s.  It would be good if automation helped, but it didn't!
* [x] instance `SeminormedRing (α →ᵇ R)`, error
  A: Gave explicitly the `Ring` instance that was proven earlier: why did it not synthesize it on its own?
* [x] instance `NormedRing (α →ᵇ R)`, error
  A: Gave explicitly the `Ring` instance that was proven earlier: why did it not synthesize it on its own?
* [x] instance `NormedAlgebra 𝕜 (α →ᵇ γ)`, synthesized type class instance is not definitionally equal to expression inferred by typing rules
  A: Gave explicitly the `Algebra` instance that was proven earlier: why did it not synthesize it on its own?

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by int-y1** [comment]
I started working on this

**Comment by int-y1** [comment]
I stopped working on this

**Comment by adomani** [comment]
See this [Zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews), regarding commit [f0737824d68e99630240f753cf43ea9e2ed58220](https://github.com/leanprover-community/mathlib4/commit/f0737824d68e99630240f753cf43ea9e2ed58220) and whether it is a good or bad idea to guide Lean into providing the missing instances.

File: Mathlib/Topology/ContinuousFunction/Bounded.lean [hunk 1]
```diff
@@ -0,0 +1,1627 @@
...
+
+section One
+
+variable [TopologicalSpace α] [PseudoMetricSpace β] [One β]
+
+@[to_additive]
+instance : One (α →ᵇ β) :=
+  ⟨const α 1⟩
+
+@[to_additive (attr := simp)] -- Porting note: Originally `@[simp, to_additive]`
```
Comment (dupuisf) [inline Mathlib/Topology/ContinuousFunction/Bounded.lean:629]:
```suggestion
@[to_additive (attr := simp)]
```
There is no need to put a porting note on this; it's just the standard fix, and we don't have to revisit it afterwards.

File: Mathlib/Topology/ContinuousFunction/Bounded.lean [hunk 2]
```diff
@@ -0,0 +1,1627 @@
...
+
+instance : SeminormedRing (α →ᵇ R) :=
+  { show Ring (α →ᵇ R) from ring,  -- porting note: this was not present in the original
+    BoundedContinuousFunction.nonUnitalSeminormedRing with }
+
+end Seminormed
+
+instance [NormedRing R] : NormedRing (α →ᵇ R) :=
+  { show Ring (α →ᵇ R) from ring,  -- porting note: this was not present in the original
+    BoundedContinuousFunction.nonUnitalNormedRing with }
```
Comment (dupuisf) [inline Mathlib/Topology/ContinuousFunction/Bounded.lean:1313]:
```suggestion
instance : SeminormedRing (α →ᵇ R) :=
  { show Ring (α →ᵇ R) from inferInstance,  -- porting note: this was not present in the original
    BoundedContinuousFunction.nonUnitalSeminormedRing with }

end Seminormed

instance [NormedRing R] : NormedRing (α →ᵇ R) :=
  { show Ring (α →ᵇ R) from inferInstance,  -- porting note: this was not present in the original
    BoundedContinuousFunction.nonUnitalNormedRing with }
```
This works, so Lean does find the instance. I guess it just doesn't try to fill in the missing fields by type class inference anymore without being explicitly told to do so.

File: Mathlib/Topology/ContinuousFunction/Bounded.lean [hunk 3]
```diff
@@ -0,0 +1,1627 @@
...
+set_option linter.uppercaseLean3 false in
+#align bounded_continuous_function.C BoundedContinuousFunction.C
+
+-- porting note: named this instance, to use it in `instance : NormedAlgebra 𝕜 (α →ᵇ γ)`
+instance algebra : Algebra 𝕜 (α →ᵇ γ) :=
+  { BoundedContinuousFunction.module,
+    BoundedContinuousFunction.ring (α := α) (R := γ) with
+    toRingHom := C
+    commutes' := fun _ _ => ext fun _ => Algebra.commutes' _ _
+    smul_def' := fun _ _ => ext fun _ => Algebra.smul_def' _ _ }
```
Comment (dupuisf) [inline Mathlib/Topology/ContinuousFunction/Bounded.lean:1379]:
```suggestion
-- porting note: named this instance, to use it in `instance : NormedAlgebra 𝕜 (α →ᵇ γ)`
instance algebra : Algebra 𝕜 (α →ᵇ γ) :=
  { BoundedContinuousFunction.module,
    BoundedContinuousFunction.ring (α := α) (R := γ) with
    toRingHom := C
    commutes' := fun _ _ => ext fun _ => Algebra.commutes' _ _
    smul_def' := fun _ _ => ext fun _ => Algebra.smul_def' _ _ }
#align bounded_continuous_function.algebra BoundedContinuousFunction.algebra
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Pull request closed without merging**